documentation/modules/upgrading/con-upgrade-strategies-for-upgrading-clients.adoc

```diff
@@ -6,65 +6,43 @@
 
 = Strategies for upgrading clients
 
-The right approach to upgrading your client applications (including Kafka Connect connectors) depends on your particular circumstances.
+[role="_abstract"]
+The goal of upgrading clients is to enable them to consume new message formats produced by updated producers.
+You do this by upgrading all the consumers for a topic _before_ upgrading any of the producers.
```

Comment:
Brokers are able to handle producers and consumers running with different versions and supporting different log message versions. 
The strategies are upgrading brokers or clients first:
- If you upgrade clients first there's a chance some new features may not work as they are not supported by brokers yet. 
- If you upgrade brokers first, in the past there was a chance client would only support an older message version and it would cause brokers to downconvert records or you would have to keep an older log message version. Since the log message version has not changed in years this has not been an issue for many versions.

Reply:
I've updated without saying too much about versions, just highlighting the potential drawbacks of both.

> The goal of upgrading Kafka clients is to enable them to consume new message formats produced by updated producers. 
> The chosen upgrade strategy depends on whether you are upgrading brokers or clients first. 
> If you upgrade clients before brokers, some new features may not work as they are not yet supported by brokers. 
> However, brokers can handle producers and consumers running with different versions and supporting different log message versions.
> 
> If you upgrade brokers first, there may be a need for down-conversion or retention of older log message versions to ensure compatibility with older applications. 
> The decision to upgrade clients or brokers first depends on several factors, such as the number of applications that need to be upgraded and how much downtime is tolerable.
> 
> Therefore, it is important to carefully consider the best approach for upgrading Kafka clients and brokers to ensure a smooth transition.